plugins/mac-videotoolbox/encoder.c

```diff
@@ -1088,11 +1365,46 @@ bool obs_module_load(void)
 			(hardware_ref) ? CFBooleanGetValue(hardware_ref)
 				       : false;
 
+		VT_DICTSTR(kVTVideoEncoderList_EncoderID, id);
+
+		switch (codec_type) {
+		case kCMVideoCodecType_H264:
+			info.get_properties2 = vt_properties_h26x;
+			info.codec = "h264";
+			break;
+#ifdef ENABLE_HEVC
+		case kCMVideoCodecType_HEVC:
+			info.get_properties2 = vt_properties_h26x;
+			info.codec = "hevc";
+			break;
+#endif
+			// 422 is used as a marker for all ProRes types,
+			// since the type is stored as a profile
+		case kCMVideoCodecType_AppleProRes422:
+			info.get_properties2 = vt_properties_prores;
+			info.codec = "prores";
+			vt_add_prores_encoder_data_to_list(hardware_accelerated,
+							   codec_type, id_ref);
+			break;
+
+		case kCMVideoCodecType_AppleProRes422Proxy:
+		case kCMVideoCodecType_AppleProRes422LT:
+		case kCMVideoCodecType_AppleProRes422HQ:
+			vt_add_prores_encoder_data_to_list(hardware_accelerated,
+							   codec_type, id_ref);
+			continue;
+
+		default:
+			continue;
```

Comment:
Make sure to `bfree(id)` here if you don't use it anymore.

Reply:
`id` should not be free'd because the pointer is used in `type_data->id`. The memory should be released as part of `vt_destroy` (which it does iirc).

Reply:
It's not used in the case of ProRes 422 Proxy, 422 LT, 422 HQ as those aren't added as their own encoders (but rather used as subtypes of 422), and not used in the case of any other codec we're not using (h263, etc). In those cases (anything that has a `continue` call in this statement here) it needs to be freed.

Reply:
Right.. in that case it really needs to be free'd.

I'd probably rather call the macro after the switch statement and just use `CFStringRef id_ref = CFDictionaryGetValue(encoder_dict, kVTVideoEncoderList_EncoderID);` for ProRes directly, but I'll leave that to Apple.